ui/lib/kv/addon/components/page/secret/details.hbs

```diff
@@ -18,6 +18,81 @@
     {{/unless}}
   </:toolbarFilters>
   <:toolbarActions>
+    {{! Start of delete options }}
+    {{#if (and @secret.deletionTime @secret.canUndelete (not @secret.destroyed))}}
+      <button
+        type="button"
+        class="toolbar-link"
+        {{on "click" (fn this.handleDelete "undelete")}}
```

Comment:
I think this should be a separate function which would clean up the `handleDelete` code